Dockerfile

```diff
@@ -1,80 +1,38 @@
-# To stop the process:
-#   > docker kill <containerid>
-##########
+# Build the base DHIS2 image
+#
 
-##########
-# BUILD STAGE 1
-# Build the DHIS2 Core server from source (Maven)
-##########
-FROM maven:3.5.3-jdk-8-slim as build
+FROM maven:3.6.2-jdk-8-slim as build
 LABEL stage=intermediate
 
 ARG IDENTIFIER=unknown
 LABEL identifier=${IDENTIFIER}
-#NB - maven-frontend-plugin breaks on Alpine linux, so we use Debian Slim instead
-#NB - maven-surefire-plugin fails with maven:3.5.4-jdk-8-slim and later.
-#     This is a recent issue possibly traced to an OpenJDK bug - https://github.com/carlossg/docker-maven/issues/90
 
 RUN apt-get update && \
+    apt-get upgrade -y && \
     apt-get install --no-install-recommends -y \
-        git=1:2.11.0-3+deb9u4 && \
+        git && \
     rm -rf /var/lib/apt/lists/*
 
-#NB - web-apps build uses `git rev-parse` to tag the build, so just copy over the whole tree for now
-COPY . /src
-
-# TODO: We should be able to achieve much faster incremental builds and cached dependencies using
-#   a wrapper build script and intelligent Docker layer caching, but for now just naively build everything
-RUN mvn clean install -T1C -f /src/dhis-2/pom.xml -DskipTests
-RUN mvn clean install -T1C -U -f /src/dhis-2/dhis-web/pom.xml -DskipTests
-
-##########
-# BUILD STAGE 2
-# Serve the packaged .war file (Tomcat)
-# Use a second build stage in a fresh container so we don't bloat it with the Maven build tools
-##########
-FROM tomcat:8.5.46-jdk8-openjdk-slim as serve
-
-ENV DHIS2_HOME=/DHIS2_home
-
-COPY docker-entrypoint.sh /usr/local/bin/
+WORKDIR /src
 
-RUN rm -rf /usr/local/tomcat/webapps/* && \
-    mkdir /usr/local/tomcat/webapps/ROOT && \
-    chmod +rx /usr/local/bin/docker-entrypoint.sh && \
-    mkdir $DHIS2_HOME && \
-    adduser --system --disabled-password --group tomcat && \
-    echo 'tomcat' >> /etc/cron.deny && \
-    echo 'tomcat' >> /etc/at.deny
+# NB: web-apps build uses `git rev-parse` to tag the build, so just copy over the whole tree for now
+COPY dhis-2 .
 
-RUN apt-get update && \
-    apt-get install --no-install-recommends -y \
-        util-linux \
-        bash \
-        unzip
+# TODO: We should be able to achieve much faster incremental builds and cached dependencies using
+RUN mvn clean install -T1C -f pom.xml -DskipTests
+RUN mvn clean install -T1C -U -f dhis-web/pom.xml -DskipTests
 
-COPY server.xml /usr/local/tomcat/conf
-COPY --from=build /src/dhis-2/dhis-web/dhis-web-portal/target/dhis.war /usr/local/tomcat/webapps/ROOT.war
+RUN cp dhis-web/dhis-web-portal/target/dhis.war /dhis.war && \
+    cd / && \
+    sha256sum dhis.war > /sha256sum.txt && \
+    md5sum dhis.war > /md5sum.txt
 
-ENTRYPOINT ["/usr/local/bin/docker-entrypoint.sh"]
+#
+# Slim final image that has the build artifacts at root-level
+#
 
-CMD ["catalina.sh", "run"]
+FROM alpine:latest
+COPY --from=build /dhis.war /dhis.war
+COPY --from=build /sha256sum.txt /sha256sum.txt
+COPY --from=build /md5sum.txt /md5sum.txt
```

Comment:
Maybe we should put the artifacts under `/srv/dhis2`?

http://www.tldp.org/LDP/Linux-Filesystem-Hierarchy/html/srv.html